pkg/storage/unified/client.go

```diff
@@ -153,3 +159,54 @@ func newResourceClient(conn *grpc.ClientConn, cfg *setting.Cfg, features feature
 	}
 	return resource.NewRemoteResourceClient(tracer, conn, clientCfgMapping(grpcutils.ReadGrpcClientConfig(cfg)), cfg.Env == setting.Dev)
 }
+
+// grpcConn creates a new gRPC connection to the provided address.
+func grpcConn(address string, reg prometheus.Registerer) (*grpc.ClientConn, error) {
+	// This works for now as the Provide function is only called once during startup.
+	// We might eventually want to tight this factory to a struct for more runtime control.
+	metrics := clientMetrics{
+		requestDuration: promauto.With(reg).NewHistogramVec(prometheus.HistogramOpts{
+			Name:    "resource_server_client_request_duration_seconds",
+			Help:    "Time spent executing requests to the resource server.",
+			Buckets: prometheus.ExponentialBuckets(0.008, 4, 7),
+		}, []string{"operation", "status_code"}),
+	}
+
+	// Report gRPC status code errors as labels.
+	unary, stream := instrument(metrics.requestDuration, middleware.ReportGRPCStatusOption)
+
+	// Set the defaults that are normally set by Config.RegisterFlags to 100MiB.
+	cfg := grpcclient.Config{
```

Comment:
You may want to call `flagext.DefaultValues(&cfg)` to get all default values.

Reply:
Sweet, changing it 👍